cmd/tink-worker/action.go

```diff
@@ -43,79 +57,81 @@ func executeAction(ctx context.Context, action *pb.WorkflowAction, wfID string)
 		timeCtx, cancel = context.WithTimeout(context.Background(), 1*time.Hour)
 	}
 	defer cancel()
-	//run container with timeout context
-	//startedAt := time.Now()
-	err = runContainer(timeCtx, id)
+	err = runContainer(timeCtx, l, id)
 	if err != nil {
-		return "Failed to run container", 1, errors.Wrap(err, "DOCKER RUN")
+		return pb.ActionState_ACTION_IN_PROGRESS, errors.Wrap(err, "DOCKER RUN")
 	}
 
 	failedActionStatus := make(chan pb.ActionState)
 
 	//capturing logs of action container in a go-routine
 	go captureLogs(ctx, id)
 
-	status, err, werr := waitContainer(timeCtx, id)
-	if werr != nil {
-		rerr := removeContainer(ctx, id)
+	status, err := waitContainer(timeCtx, id)
+	if err != nil {
+		rerr := removeContainer(ctx, l, id)
 		if rerr != nil {
-			log.WithField("container_id", id).Errorln("Failed to remove container as ", rerr)
+			l.With("containerId", id)
+			l.Error(errors.Wrapf(rerr, errRemoveContainer))
 		}
-		return "Failed to wait for completion of action", status, errors.Wrap(err, "DOCKER_WAIT")
+		return status, errors.Wrap(err, "DOCKER_WAIT")
 	}
-	rerr := removeContainer(ctx, id)
+	rerr := removeContainer(ctx, l, id)
 	if rerr != nil {
-		return "Failed to remove container of action", status, errors.Wrap(rerr, "DOCKER_REMOVE")
+		return status, errors.Wrap(rerr, "DOCKER_REMOVE")
 	}
-	log.Infoln("Container removed with Status ", pb.ActionState(status))
+	l = l.With(infoContainerRemoved, status.String())
+	l.Info()
 	if status != pb.ActionState_ACTION_SUCCESS {
 		if status == pb.ActionState_ACTION_TIMEOUT && action.OnTimeout != nil {
-			id, err = createContainer(ctx, action, action.OnTimeout, wfID)
+			id, err = createContainer(ctx, l, action, action.OnTimeout, wfID)
 			if err != nil {
-				log.Errorln("Failed to create container for on-timeout command: ", err)
+				l.Error(errors.Wrapf(err, errCreateContainer))
 			}
-			log.Infoln("Container created with on-timeout command : ", action.OnTimeout)
+			l := l.With(infoContainerCreated, action.GetOnTimeout())
+			l.Info()
 			failedActionStatus := make(chan pb.ActionState)
 			go captureLogs(ctx, id)
 			go waitFailedContainer(ctx, id, failedActionStatus)
-			err = runContainer(ctx, id)
+			err = runContainer(ctx, l, id)
 			if err != nil {
-				log.Errorln("Failed to run on-timeout command: ", err)
+				l.Error(errors.Wrapf(err, errFailedToRunCmd))
 			}
 			onTimeoutStatus := <-failedActionStatus
-			log.Infoln("On-Timeout Container status : ", onTimeoutStatus)
+			l.Info(fmt.Sprintf(infoContainerStatus, "timeout", onTimeoutStatus))
```

Comment:
ditto, not sure what this log message is trying to relay though: `infoContainerStatus   = "on %s container status: %v"` wrong `info*` var?

Reply:
Redefined the key there.